pkg/apis/kops/validation/validation.go

```diff
@@ -855,3 +860,40 @@ func validateNodeLocalDNS(spec *kops.ClusterSpec, fldpath *field.Path) field.Err
 
 	return allErrs
 }
+
+func validateServiceOIDCProvider(c *kops.ClusterSpec, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	thumbprintFormat := regexp.MustCompile(`^[a-fA-F0-9]{40}$`)
+	provider := c.ServiceOIDCProvider
+
+	if kops.CloudProviderID(c.CloudProvider) != kops.CloudProviderAWS {
+		allErrs = append(allErrs, field.Forbidden(path, "serviceOIDCProvider is supported only in AWS"))
+	}
+
+	if provider.Issuer == "" {
+		allErrs = append(allErrs, field.Required(path.Child("issuer"), ""))
+	} else {
+		// Based on these requirements https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html#manage-oidc-provider-console
+		url, err := url.ParseRequestURI(provider.Issuer)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(path.Child("issuer"), provider.Issuer, "must be a URL"))
+		}
+		if url.Scheme != "https" {
```

Comment:
Scheme and port should be in an else block. They should not cascade if the URL didn't parse.